### ISSUE TITLE
Memory leak for dispatcher

### ISSUE THREAD

**Issue by flamurey** [ngrx/store#167]
Hi,

I don't now why but my app have memory leak. By using chrome I detect into heap dump what dispatcher undicrectly contains all actions with payloads. My app frequently load large data and dispatching it.

![image](https://cloud.githubusercontent.com/assets/12752623/16655803/bbf3bae6-4464-11e6-92fd-8fb17813e0e5.png)

**Comment by MikeRyanDev** [comment]
Interesting. Do you think you might be able to reproduce it in a Plunker or in a separate repo?

**Comment by flamurey** [comment]
https://plnkr.co/edit/pzODIm

watching memory profiler you will see as used memory increasing

**Comment by asarode** [comment]
Has any progress been made on this?

**Comment by dchacke** [comment]
+1. I have been running into issues with memory as well and am curious if it has to do with this.

**Comment by MikeRyanDev** [comment]
I originally left this issue open because we were using a custom built subject for tracking actions. Recent versions of @ngrx/store are just using a `BehaviorSubject` and the `queue` scheduler. If this memory problem still persists, this is most likely a bug in rxjs. Thanks!

**Comment by michaelcleary** [comment]
I'm using rxjs/store (2.2.1) in an application that puts some fairly large result sets into the store.  Unfortunately because of this memory leak the application crashes the browser as the store after a few data requests.  If the bug is with rxjs, is it possible something could be done to work around it in ngrx?

**Comment by mfahlen** [comment]
I'm also seeing memory leaks using @ngrx/store version 2.2.1. It looks like dispatched actions aren't garbage collected. Any help looking into this issue would be greatly appreciated.

**Comment by MikeRyanDev** [comment]
@michaelcleary @mfahlen 

I have two theories:
1. Are you using @ngrx/store-devtools? The devtools are going to keep all actions to make recomputing of states possible. There is an option to only retain the last `n` actions if this is case.
2. Are you using ngrx-store-logger? If you are using ngrx-store-logger (or something similar to log actions to the console) then the actions won't be able to be garbage collected until the console is cleared.

Let me know if this is not the case and I'll reopen the issue.

**Comment by michaelcleary** [comment]
No, did initially think that could be the cause, so took both those out and
found it was exactly the same.  Even with the example app you can see that
the heap is steadily growing with each request and this is apparently
because the _subscriptions array inside _dispatcher retains every
QueueAction (as shown in the original post).

**Comment by MikeRyanDev** [comment]
@blesh Any ideas? Code path that reproduces this:

```ts
const dispatcher = new BehaviorSubject({ type: 'INIT' });

dispatcher.observeOn(queue).subscribe();
```

Previous actions pushed into the `dispatcher` subject won't be released anymore.

**Comment by josh-sachs** [comment]
I believe I am also being affected by this issue. It is quite severe. Can anyone recommend a previous version of ngrx/store that isn't impacted by this while a resolution is found?

**Comment by MikeRyanDev** [comment]
For reference, I've opened up an issue about this on the rxjs repo: reactivex/rxjs#2244

@kudoz83 There isn't a previous version of @ngrx/store unaffected by this issue. Observing the action stream on a queue prevents action reentrancy.

**Comment by MikeRyanDev** [comment]
Fixed in rxjs v5.0.3
